src/libos/src/fs/file_table.rs

```diff
@@ -236,4 +253,18 @@ impl FileTableEntry {
     pub fn set_close_on_spawn(&mut self, close_on_spawn: bool) {
         self.close_on_spawn = close_on_spawn;
     }
+
+    pub fn set_table_id(&mut self, table_id: u64) {
+        self.table_id = table_id;
+    }
+}
+
+impl Drop for FileTableEntry {
+    fn drop(&mut self) {
+        // Remove the advisory locks if necessary
+        if let Ok(inode_file) = self.file.as_inode_file() {
+            trace!("drop file table entry of {:?}", inode_file);
+            inode_file.remove_advisory_locks(self.table_id).unwrap();
+        }
+    }
```

Comment:
The proposed changes made to `FileTable` makes it much more complex that the original intention of `FileTable`, which has a simple task to maintain the map between fd and files for a process. The proposed changes put more burden to `FileTable`, resulting in a tight coupling between file tables and advisory locks. Could you find a way to extract the logic of removing file locks out of file tables?

Reply:
Done, Process have to handle the closing files, exiting process, or executing binary to release file locks manually.